src/Symfony/Component/Security/Csrf/TokenGenerator/UriSafeTokenGenerator.php

```diff
@@ -47,7 +41,10 @@ class UriSafeTokenGenerator implements TokenGeneratorInterface
      */
     public function __construct(SecureRandomInterface $random = null, $entropy = 256)
     {
-        $this->random = $random ?: new SecureRandom();
+        if (null !== $random) {
+            @trigger_error('The $random parameter in '.__METHOD__.' is deprecated since version 2.8 and will be removed in 3.0.', E_USER_DEPRECATED);
+        }
```

Comment:
We should make sure that we are able to drop the first argument in Symfony 3.0 and only need to pass the entropy. For this, we need to drop the type hint for the `SecureRandomInterface`:

```php
public function __construct($random, $entropy = 256)
{
    if ($random instanceof SecureRandomInterface || func_num_args() === 2) {
        @trigger_error('The '.__METHOD__.' method now requires the entropy to be given as the first argument. The SecureRandomInterface will be removed in 3.0.');

        $this->entropy = $entropy;
    } elseif (func_num_args() === 0) {
        $this->entropy = 256;
    } else {
        $this->entropy = $random;
    }
}
```